evaluate-default-paths.yml: Add paths for wasi

radical opened the pull request:
.. and exclude wasm specific paths from coreclr.

- This will ensure that wasm specific changes don't trigger all the coreclr, and then mono builds
- And changes in new wasi specific files will trigger wasm jobs
- And don't trigger wasm debugger tests on manual pipelines if they are run on `runtime` pipeline

radical committed:
evaluate-default-paths.yml: Add paths for wasi

.. and exclude wasm specific paths from coreclr.

radical committed:
disable runtime-wasm-optional automatic trigger for PRs

radical committed:
CI: skip debugger tests on runtime-wasm if they run on runtime pipeline

.. too.

radical committed:
fix condition

kg commented on eng/pipelines/common/evaluate-default-paths.yml:
```diff
@@ -7,9 +7,13 @@ parameters:
   _const_paths:
     _wasm_specific_only: [
         eng/testing/ProvisioningVersions.props
+        eng/testing/Was*Runner*
```
`(m|i)` instead of `*`?

kg replied:
or use `?` to be clear that it's one character

kg replied:
with the caveat that i don't know what kind of glob/regex syntax is being used here :-)

radical replied:
Fixed. these are used with `git diff`, and follow `git pathspec`.

radical replied:
hmph azdo didn't like `?`: `/eng/pipelines/common/evaluate-default-paths.yml (Line: 10, Col: 24): While parsing a flow sequence, did not find expected ',' or ']'.`

radical replied:
Took the easy way out with `WasmRunner*`, and `WasiRunner`.

kg commented on eng/pipelines/common/templates/wasm-debugger-tests.yml:
```diff
@@ -4,6 +4,7 @@ parameters:
   isWasmOnlyBuild: false
   browser: 'chrome'
   shouldContinueOnError: false
+  runOnlyOnWasmPipelines: false
```
I'm confused as to why this would want to run on non-wasm pipelines. Are the wasm debugger tests not a wasm pipeline?

radical replied:
it should really be named `runOnlyOnWasmOnlyPipelines` or such!

radical replied:
So, it should run only on runtime-wasm* pipelines which have only wasm builds, and are run manually.

radical committed:
address review feedback from @kg

radical committed:
Merge remote-tracking branch 'origin/main' into ci-more-fixes

radical committed:
Fix yml issue

Using `Was?Runner*` failed to validate on azdo with:

`/eng/pipelines/common/evaluate-default-paths.yml (Line: 10, Col: 24): While parsing a flow sequence, did not find expected ',' or ']'.`

radical committed:
fix syntax error in yml

radical commented:
This is a CI only change, and the failures are unrelated.
